integration-tests/conftest.py

```diff
@@ -15,7 +15,12 @@ def pytest_addoption(parser):
         help="",
         default="http://0.0.0.0:3334",
     )
-
+    parser.addoption(
+        "--ws_url",
+        action="store",
+        help="",
+        default="ws://0.0.0.0:8354",
```

Comment:
This should be localhost. 

`0.0.0.0` means "any ip", which is used by servers to define which local addresses to listen. 
For example, a server listening to `127.0.0.1`, would only allow connections from localhost. Server listening for `192.168.0.42` (if the host has such an IP address), would only allow connections for this network interface (and connections localhost will be denied).

Listening to `0.0.0.0` means "allow connections from any network interface".
But it makes no sense to set it on a client side, this should be a specific address. Localhost in this case.

```suggestion
        default="ws://127.0.0.1:8354",
```